### ISSUE TITLE
Problem on generate thumbnails for PDFs or PPTXs

### ISSUE THREAD

**Issue by haozeng0** [samvera-labs/hyku#1193]
Freshly installed in docker, couldn't generate thumbnails for PDF or PPTX files. Images work fine.

Here is the hyku_workers log:

 I, [2017-06-02T14:54:16.082386 #1]  INFO -- : [ActiveJob] [CreateDerivativesJob] [08e84faf-a3dd-47f3-baf5-4b372c32369e] Performing CreateDerivativesJob (Job ID: 08e84faf-a3dd-47f3-baf5-4b372c32369e) from Sidekiq(default) with arguments: #<GlobalID:0x0055b566519ea8 @uri=#<URI::GID gid://hyku/FileSet/afe7f62f-e740-4691-99ef-4adf1c9779c9>>, "afe7f62f-e740-4691-99ef-4adf1c9779c9/files/bd9db74f-5d75-4a1e-a8b6-d6707b86d44a", "/data/tmp/uploads/e9753484-1371-4e1f-aa73-c51045c989d1/hyrax/uploaded_file/file/3/EDS_VuFind_Integration_Filipe_Bento_EBSCO_Sept2015.pptx"
 **E, [2017-06-02T14:54:18.644273 #1] ERROR -- : [ActiveJob] [CreateDerivativesJob] [08e84faf-a3dd-47f3-baf5-4b372c32369e] Error performing CreateDerivativesJob (Job ID: 08e84faf-a3dd-47f3-baf5-4b372c32369e) from Sidekiq(default) in 2561.49ms: MiniMagick::Invalid (`identify /tmp/mini_magick20170602-1-1rpog0s.pdf` failed with error:
 ):**
 /usr/local/bundle/gems/mini_magick-4.7.0/lib/mini_magick/image.rb:203:in `rescue in validate!'
 /usr/local/bundle/gems/mini_magick-4.7.0/lib/mini_magick/image.rb:201:in `validate!'
 /usr/local/bundle/gems/mini_magick-4.7.0/lib/mini_magick/image.rb:113:in `block in create'
 /usr/local/bundle/gems/mini_magick-4.7.0/lib/mini_magick/image.rb:112:in `tap'
 /usr/local/bundle/gems/mini_magick-4.7.0/lib/mini_magick/image.rb:112:in `create'
 /usr/local/bundle/gems/mini_magick-4.7.0/lib/mini_magick/image.rb:34:in `read'
 /usr/local/bundle/gems/mini_magick-4.7.0/lib/mini_magick/image.rb:90:in `block in open'

**Comment by mjgiarlo** [comment]
Thanks for reporting this, @haozeng0. I've added this near the top of our backlog.

I wonder if we need to install ghostscript (if we aren't already).

**Comment by mjgiarlo** [comment]
#1194 should have fixed PDFs. This issue now focuses on office document derivatives.

**Comment by mjgiarlo** [comment]
The thumbnails for office documents (tested with .ppt and .pptx) render as a broken image with the text "Default" within it. The thumbnail links look like https://troubleshooting.demo.hydrainabox.org/images/0c6e5278-9a68-4489-8745-e65f7dce8c98%2Ffiles%2F938ff88f-1030-410d-9679-38d55ca07e22/full/!150,300/0/default.jpg

I have seen this in both the docker and AWS builds. Have not tested this locally or in Hyrax.

**Comment by mjgiarlo** [comment]
From @jcoyne:

> Seeing this on AWS:
```
RuntimeError: Unable to execute command "soffice --invisible --headless --convert-to pdf 
--outdir /tmp /var/app/current/tmp/uploads/83/09/2d/c2/83092dc2-22f0-4ddf-
a72f-e06e36a3bc5c/83092dc2-22f0-4ddf-a72f-e06e36a3bc5c20170602-25338-1gjs6r5". Exit
code: pid 2586 exit 77 Error message: [Java framework] Error in function createSettingsDocument
(elements.cxx). javaldx failed! Warning: failed to read path from javaldx 
```
> Apparently `soffice` needs a writable directory in which to write its configs.

**Comment by dlpierce** [comment]
In Docker I'm not seeing the above error about needing a writable config directory. `soffice` seems to run properly.

After fixing hydra-derivatives so it reads the solr uri properly, it seems the broken thumbnail images come from riiif trying to use the original file to create the thumbnail, but it of course doesn't know how to read a .ppt. I'll continue working on this.

**Comment by dlpierce** [comment]
Fixed by #1280
